src/elements/content-sidebar/AddTaskButton.js

```diff
@@ -25,17 +26,29 @@ type Props = {| ...AddTaskButtonProps, ...PassThroughProps |};
 
 type State = {
     isTaskFormOpen: boolean,
+    taskType: ?TaskType,
 };
 
 class AddTaskButton extends React.Component<Props, State> {
     state = {
         isTaskFormOpen: false,
+        taskType: null,
     };
 
     static defaultProps = {
         isDisabled: false,
     };
 
+    getMessageForModalTitle(taskType: ?TaskType): Object {
+        switch (taskType) {
```

Comment:
Should `TASK_TYPE_GENERAL` be the default case if `taskType` is not provided?

Reply:
the backend creates an approval task if `taskType` is not provided

Reply:
we can the type non-optional/non-nullable here too